core_context_impl/src/test/java/com/google/instrumentation/trace/ContextSpanHandlerImplTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.instrumentation.common.NonThrowingCloseable;
+import io.grpc.Context;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+/** Unit tests for {@link ContextSpanHandlerImpl}. */
+@RunWith(JUnit4.class)
+public class ContextSpanHandlerImplTest {
+  private static final Tracer tracer = Tracer.getTracer();
+
+  @Mock private Span span;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  public void getCurrentSpan_WhenNoContext() {
+    assertThat(new ContextSpanHandlerImpl().getCurrentSpan()).isNull();
+    assertThat(tracer.getCurrentSpan()).isSameAs(BlankSpan.getInstance());
+  }
+
+  @Test
+  public void getCurrentSpan() {
+    Context origContext =
+        Context.current().withValue(GrpcTraceUtils.getContextSpanKey(), span).attach();
+    assertThat(tracer.getCurrentSpan()).isSameAs(span);
+    Context.current().detach(origContext);
```

Comment:
I still think that the test should restore the context in a finally block, as an example for users, and to prevent a failure in this test from causing other tests to fail.